internal/common/function/function.go

```diff
@@ -157,6 +167,82 @@ func ValidateRestoreSpec(ctx context.Context, clientInstance client.Client, nonA
 	return nil
 }
 
+// ValidateBslSpec return nil, if NonAdminRestore is valid; error otherwise
+func ValidateBslSpec(ctx context.Context, clientInstance client.Client, nonAdminBsl *nacv1alpha1.NonAdminBackupStorageLocation, enforcedBslSpec *velerov1.BackupStorageLocationSpec) error {
+	if nonAdminBsl.Spec.BackupStorageLocationSpec.Credential == nil {
+		return fmt.Errorf("NonAdminBackupStorageLocation spec.bslSpec.credential is not set")
+	} else if nonAdminBsl.Spec.BackupStorageLocationSpec.Credential.Name == constant.EmptyString || nonAdminBsl.Spec.BackupStorageLocationSpec.Credential.Key == constant.EmptyString {
+		return fmt.Errorf("NonAdminBackupStorageLocation spec.bslSpec.credential.name or spec.bslSpec.credential.key is not set")
+	}
+
+	var enforcedSpec reflect.Value
```

Comment:
asked in slack to redo this part the same way as NAB and NAR

Reply:
correct and my answer is that it can't be the same as here we are dealing with values from ConfigMap which is not the case in other objects, so I am not sure how this can be same?

https://github.com/migtools/oadp-non-admin/pull/139/files/8c8df21394a0a3ea3513d5346ded0c040bfc82c3#diff-de455907ed1c5413fa6248ebea483af559585f8f6277a7c8b370813b9bb254f9R199-R218

Reply:
the type of the object does not require new implementation

both NAB and NAR also have `map[string]string` fields 

https://github.com/migtools/oadp-non-admin/blob/019bb2a677d4648cce1443966d6686cd46a7cef4/internal/common/function/function_test.go#L285-L289

https://github.com/migtools/oadp-non-admin/blob/019bb2a677d4648cce1443966d6686cd46a7cef4/internal/common/function/function_test.go#L595-L601

Reply:
I'll look into that.

Reply:
you need to pass a `enforcedSpec` to the tests. Function do not expect it to be nil. At max, it will be a empty struct

adding here https://github.com/mpryc/oadp-non-admin/blob/nonadminstoragelocation_controller-v2/internal/common/function/function_test.go#L758

`enforcedSpec: &velerov1.BackupStorageLocationSpec{},` should fix teh panic, for example

Reply:
Are we 100% positive that structure will always be present in the running controller code? I mean if that empty structure is present even if no enforce settings are set by admin ?

Reply:
yes

https://github.com/migtools/oadp-non-admin/blob/019bb2a677d4648cce1443966d6686cd46a7cef4/cmd/main.go#L201-L213

it is initialized as a pointer to a empty struct

if admin user did set any enforcement, it is updated to a pointer to a non empty struct

Reply:
again, what you are trying to accomplish goes against design https://github.com/migtools/oadp-non-admin/blob/master/docs/design/admin_control_over_spec.md?plain=1#L27

if you want to change this, you also need to change backup and restore enforcement

Reply:
At the time of writing that design I was not aware of need of the config. I think we must change that. I am not sure if we want to change it for Backup/Restore - if yes I would say separate PR

Reply:
BSL config spec is a bit different that the other non-nested typed spec levels. If we want to revisit backup/restore enforcement specs then lets just make note/create issue for it and move on. It can be a seperate PR. The changes in the current PR cater to NaBSL, lets make it rock solid and keep moving :)

Reply:
Is the below Issue which will need to be revisited solving the debate within this PR or is change to this PR still required?

https://github.com/migtools/oadp-non-admin/issues/151

Reply:
We can review and revisit this in #151